File: mcorg.backends/project-service/src/main/java/app/mcorg/organizer/presentation/rest/entities/project/ProjectResponse.java
package app.mcorg.organizer.presentation.rest.entities.project;

import app.mcorg.organizer.domain.model.project.Project;
import org.springframework.lang.NonNull;

import java.util.List;

public record ProjectResponse(@NonNull String id,
                              @NonNull String name,
                              @NonNull boolean isArchived,
                              @NonNull List<TaskResponse> tasks,
                              @NonNull List<CountedTaskResponse> countedTasks) {
    public static ProjectResponse from(Project project) {
        return new ProjectResponse(
                project.id(),
                project.name(),
                project.isArchived(),
                project.tasks().stream()
                        .map(TaskResponse::from)
                        .toList(),
                project.countedTasks().stream()
                        .map(CountedTaskResponse::from)
                        .toList()
        );
    }
}

File: mcorg.frontends/organizer-webapp/api/projects/models/ProjectResponse.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CountedTaskResponse } from './CountedTaskResponse';
import type { TaskResponse } from './TaskResponse';

export type ProjectResponse = {
  id: string;
  name: string;
  isArchived: boolean;
  tasks: Array<TaskResponse>;
  countedTasks: Array<CountedTaskResponse>;
};
